packages/world/test/KeysWithValueModule.t.sol

```diff
@@ -208,4 +208,79 @@ contract KeysWithValueModuleTest is Test {
     assertEq(keysWithValue[0], key1);
     assertEq(keysWithValue[1], key2);
   }
+
+  function testGetKeysWithValueMany10() public {
+    _installKeysWithValueModule();
+
+    uint256 amount = 10;
+    uint256 value = 1;
+
+    bytes32[] memory lastKey = new bytes32[](1);
+    lastKey[0] = bytes32(uint256(amount - 1));
+
+    for (uint256 i; i < amount - 1; i++) {
+      bytes32[] memory key = new bytes32[](1);
+      key[0] = bytes32(i);
+      world.setRecord(namespace, sourceName, key, abi.encodePacked(value));
+    }
+
+    // !gasreport Setting the last of 10 keys
+    world.setRecord(namespace, sourceName, lastKey, abi.encodePacked(value));
+
+    // !gasreport Get list of 10 keys with a given value
```

Comment:
It would be good if the gas report value was a parameter and not hardcoded in a comment - because I need to make 3 of these functions.

Reply:
yep, for now you could try putting
```solidity
uint256 amount = 100;
uint256 value = 1;

bytes32[] memory lastKey = new bytes32[](1);
lastKey[0] = bytes32(uint256(amount - 1));

for (uint256 i; i < amount - 1; i++) {
  bytes32[] memory key = new bytes32[](1);
  key[0] = bytes32(i);
  world.setRecord(namespace, sourceName, key, abi.encodePacked(value));
}
```
in a helper to reduce the issue of repetition

Reply:
Oh yup. Even that that would be better, thanks.